### PR TITLE
Add missing free context in at the end of aes_crypt_xts_size()

### LINKED ISSUE #4176: Missing context freeing operation at the end of aes_crypt_xts_size
### Description
- Type: Bug 
- Priority: Minor

---------------------------------------------------------------
## Bug

**OS**  
Mbed OS|linux|windows|

**mbed TLS build:**  
Version: 2.25.0 / git commit  55d97f2 

In function "void aes_crypt_xts_size( int size, int retval )" in tests/suite/test_suite_aes.function the aes_xts context is initialized but never freed. The following lines should be added:
exit:
    mbedtls_aes_xts_free(&ctx);

Spotted since in the alternate implementation we have running under ucLinux on an NXP Kinetis K61, which makes usage of the mmcau hardware accelerator, we open a device when the context is initialized and the device was never freed.

### PR DESCRIPTION
**Pull request by JoeSubbiani**
## Description
in file tests/suite/test_suite_aes.function, aes_crypt_xts_size()
did not free the context upon the function exit.
The function now frees the context on exit.

This Closes #4176

## Status
**READY**

## Requires Backporting
Yes 
[development_2.x](https://github.com/ARMmbed/mbedtls/pull/4678)
[mbedtls-2.16](https://github.com/ARMmbed/mbedtls/pull/4684)

## Migrations
If there is any API change, what's the incentive and logic for it.

NO

## Todos
- [x] Changelog
- [x] Backported: [2.x](https://github.com/ARMmbed/mbedtls/pull/4678), [mbedtls-2.16](https://github.com/ARMmbed/mbedtls/pull/4684)

## Steps to test or reproduce
test_suite_aes

### CODE REVIEW

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Ok for the code change.

We don't always write changelog entries for bug fixes in the test suites, but given that this bug was externally reported, I think we should. Please add a file under `ChangeLog.d` with a changelog entry explaining the bug from a library user's point of view (a resource was never freed when running one particular test suite with an alternative AES implementation) and referencing the issue number — see existing changelog entries for examples.

**Comment by gilles-peskine-arm** [comment]
The issue is present in 2.16 as well, so we need a second backport.

File: ChangeLog.d/issue4176.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Fix an issue where resource is never freed when running one particular
+     test suite with an alternative AES implementation. Fixes #4176
```
Comment (daverodgman) [inline ChangeLog.d/issue4176.txt:3]: Missing full stop.

**Review by daverodgman — CHANGES_REQUESTED** [review]

File: tests/suites/test_suite_aes.function [hunk 2]
```diff
@@ -207,6 +207,8 @@ void aes_crypt_xts_size( int size, int retval )
     memset( data_unit, 0x00, sizeof( data_unit ) );
 
     TEST_ASSERT( mbedtls_aes_crypt_xts( &ctx, MBEDTLS_AES_ENCRYPT, length, data_unit, src, output ) == retval );
+exit:
+    mbedtls_aes_xts_free(&ctx);
```
Comment (daverodgman) [inline tests/suites/test_suite_aes.function:211]: Minor: improper spacing.

**Review by daverodgman — CHANGES_REQUESTED** [review]

File: ChangeLog.d/issue4176.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Fix an issue where resource is never freed when running one particular
+     test suite with an alternative AES implementation. Fixes #4176.
```
Comment (gilles-peskine-arm) [inline ChangeLog.d/issue4176.txt:3]: We're rather picky on good English in changelog entries.

* “Fix an issue where X happens” → ”Fix X“ — the extra words are just a distraction.
* “resource” → “a resource”
* “where resource is never freed” has a name: it's a resource leak
* “when running one particular test suite” → “in a test suite”

```suggestion
   * Fix a resource leak in a test suite with an alternative AES
     implementation. Fixes #4176.
```

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]

**Review by gilles-peskine-arm — APPROVED** [review]

**Review by daverodgman — APPROVED** [review]

**Pull request merged**